### DISCUSSION TITLE
When do we consider a bee as a queen?

### DISCUSSION THREAD

**Discussion by janaobsteter** [HighlanderLab/SIMplyBee#290] [Q&A] [answered]
I think we agreed that a bee can only be in the `queen` slot if it's mated --> otherwise it should be in the `virginQueens` slot. This raises a couple of questions / issues:
- why do we then have a function `isQueen()` and `isQueenMated()`?
-`isQueen()` and `isQueenMated()` can show a different thing. For example:
```
#' founderGenomes <- quickHaplo(nInd = 20, nChr = 1, segSites = 100)
#' SP <- SimParamBee$new(founderGenomes)
#' basePop <- createVirginQueens(founderGenomes)
#'
#' drones <- createDrones(x = basePop[1], nInd = 100)
#'
#' # If input is a Pop class of virgin queen(s)
#' virginQueen1 <- basePop[2]
#' isQueen(virginQueen1)
#' (matedQueen1 <- cross(
#'   x = virginQueen1,
#'   fathers = drones[1:5]
#' ))
#' 
#' isQueen(virginQueen1)
#' isQueenMated(virginQueen1)
```

returns
```
> isQueen(virginQueen1)
[1] TRUE
> isQueenMated(virginQueen1)
[1] FALSE
```

**Comment by gregorgorjanc** [comment]
`isQueen` is based on `isCaste` https://github.com/HighlanderLab/SIMplyBee/blob/e88eaef9af5fc9c10a24cbf8e691af89f9bfad83/SIMplyBee/R/Functions_L0_auxilary.R#L603, which checks the caste of an individual in SimParamBee object.

So, when you mated the virgin queen above, the cross function changed the caste for the individual in the SimParamBee object and now both the `matedQueen1` and `virginQueen1` contain the same individual whose ID is linked to the ID in the SimParamBee and associated caste, which has been set to queen.

Does this make sense?

**Comment by janaobsteter** [comment]
Yes, I do understand why it happens, it is just super confusing. A queen shouldn't be a queen if not mated - hence I don't know whether one of the functions is redundant? And second, I think we should avoid such confusing examples - we can assign the queen back to itself?

**Reply by gregorgorjanc** [reply]
Good point. I think this reflect evolution of the code. Initially we did not have isCaste/isQueen/…. We only had isQueenMated and we used it in other functions. So, I guess we should remove the isQueenMated in favour of isQueen. Can you check where we use it? Best to open an issue so we track this properly. Also, we should make sure that cross doesn’t change caste of a failed mating (say mating with 0 fathers)!!!!

**Reply by gregorgorjanc** [reply]
@janaobsteter is this resolved so we mark the discussion as answered?

**Comment by janaobsteter** [comment, marked as answer]
In SIMplyBee, the bee is a queen only when she's mated.
